aws-datastore/src/main/java/com/amplifyframework/datastore/syncengine/Merger.java

```diff
@@ -145,4 +155,22 @@
             }
         }, failure -> onNotPresent.call());
     }
+
+    /**
+     * The strategy to use while merging. Whether to consider the contents of the mutation
+     * outbox before saving data locally, or, to ignore it.
+     */
+    enum MergeStrategy {
+        /**
+         * When merging, the contents of the mutation outbox will *not* be considered.
```

Comment:
I'm not sure I see this branch in the requirements of [JS implementation](https://github.com/aws-amplify/amplify-js/blob/master/packages/datastore/src/sync/merger.ts) -- why do we ignore mutations and clobber on a network update when it seems like the intent is wait until the last mutation is applied?

Reply:
There are four sources of state in this system:
1. User's data, received from DataStoreAPI;
2. Mutation responses, received when user's data is published;
3. Subscription responses, received when any data change occurs on the remote system;
4. Sync responses, received at system startup.

3 and 4 don't have pending mutations, we just get inbound data. 1 generates a pending mutation, but doesn't process it.

2 is unique. When the mutation processor starts processing a pending mutation, it is _guaranteed_ that it is in the outbox (that's where it came from.) So by the time it gets around to merging it, we _expect_ that mutation to be in the outbox. (We haven't removed it, yet.) In fact, "clobber existing" could be more strongly implemented as "demand that it DOES exist, and fail if it doesn't".